### ISSUE TITLE
router.push by default doesn't trigger a page reload

### ISSUE THREAD

**Issue by baptisteArno** [vercel/next.js#21787]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

14.9

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**How are you deploying your application?**

Vercel

**Describe the Bug**

When pushing a new path:

```tsx
router.push("/alternate")
```

it doesn't reload the page, it pushes as if I set the `shallow` option to `true`.

**Expected Behavior**

It should reload the page by default (shallow: `false`)

**To Reproduce**

1 - Clone `with-react-native-web-app` example
2 - Replace the Link with a onClick function that calls `router.push("/alternate")`
3 - It pushes the path without browser reload.

**Comment by machineghost** [comment]
As I understand it (warning: I'm not a maintainer or affiliated at all), whether or not you use shallow, it *shouldn't* trigger a true browser refresh:

> Linking between pages

> The Next.js router allows you to do client-side route transitions between pages, similar to a single-page application.
> A React component called Link is provided to do this client-side route transition.

* https://nextjs.org/docs/routing/introduction

The concept of a single-page application is that you *don't* refresh when the user clicks a link. Instead, you just use Javascript to change the DOM, to show a new "virtual page" when the user visits it.  Developers embrace this concept because Javascript can change the DOM significantly faster than the browser can load a new page.

In other words, you have nothing to gain by *actually* reloading: it just makes your site slower.  So instead Next *fakes* a page transition ... and it *does* change the URL, so if you do refresh, you do get a different page ... but it doesn't literally load a fresh HTML page every time you click a link.

The `shallow` prop just determines whether Next re-runs the data-fetching stuff (`getStaticProps`, `getServerSideProps` or `getInitialProps`) when you follow the link.

If you actually want a reload (and again, you probably don't), there's a very simple solution: use `window.location = newUrl` instead of the Next router.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
